File: RestaurantFrontEnd/ClientApp/src/app/models/item.ts
import { Entity } from "./entity";

import { Category } from './category';
import { Unit } from './unit';
export class Item extends Entity{
  
  Name: string;
  Code: string;
  Category?: Category;
  Unit?: Unit;
  Price: number;
  Description: string; 
  Available: boolean;
  getName() {
    return this.constructor.name;
  }
  constructor(params: Item = {} as Item)
  {
    super(params);
    this.Name = params.Name;
    this.Code = params.Code;
    this.Category = params.Category;
    this.Unit = params.Unit;
    this.Price = params.Price; 
    this.Description = params.Description;
    this.Available = params.Available;
  }
}

File: RestaurantDataModel/Model/Entity/Item.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Restaurant.DataModel.Entity {
    
    public class Item : Entity, IEntity {
       
        public Item() {
			OrderedItems = new List<OrderedItem>();
        }
        [JsonSerializeAttribute]
        public virtual System.Guid Id { get; set; }
        [JsonSerializeAttribute]
        public virtual Category Category { get; set; }


        [JsonSerializeAttribute]
        [Required]

        public virtual Unit Unit { get; set; }
        [JsonSerializeAttribute]
        [StringLength(10)]
        public virtual string Code { get; set; }


        [JsonSerializeAttribute]
        [Required]
        [StringLength(50)]
        public virtual string Name { get; set; }
        [JsonSerializeAttribute]
        [Required]
        public virtual decimal Price { get; set; }
        [JsonSerializeAttribute]
        [Required]
        public virtual bool Available { get; set; }
        [JsonSerializeAttribute]
        [StringLength(2147483647)]
        public virtual string Description { get; set; }
        [JsonIgnore]
        public virtual IList<OrderedItem> OrderedItems { get; set; }
        
        
    }
}
